### PR TITLE
PN-5211 - ApiGateway configuration

### PR DESCRIPTION
**Pull request by alessandrogelmi**

### CODE REVIEW

**Review by fdammassa — APPROVED** [review]

File: codegen/config.json [hunk 1]
```diff
@@ -3,14 +3,11 @@
         {
             "intendedUsage": "B2B",
```
Comment (AndreaCimini90) [inline codegen/config.json:4]: Change this with WEB

File: docs/openapi/api-external-pn-bff.yaml [hunk 2]
```diff
@@ -17,16 +17,23 @@ servers:
...
+              schema:
+                type: string
         '500':
-          description: Internal Server Error
+          description: InternalServerError
+          content:
+            application/problem+json:
+              schema:
+                $ref: 'remote-refs.yaml#/components/schemas/Problem'
   '/template/api/httpheaders':
```
Comment (AndreaCimini90) [inline docs/openapi/api-external-pn-bff.yaml:37]: Is this needed?

File: docs/openapi/api-internal-pn-bff.yaml [hunk 3]
```diff
@@ -17,16 +17,23 @@ servers:
...
+              schema:
+                type: string
         '500':
-          description: Internal Server Error
+          description: InternalServerError
+          content:
+            application/problem+json:
+              schema:
+                $ref: 'remote-refs.yaml#/components/schemas/Problem'
   '/template/api/httpheaders':
```
Comment (AndreaCimini90) [inline docs/openapi/api-internal-pn-bff.yaml:37]: As before

File: src/main/java/it/pagopa/pn/bff/rest/StatusController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+public class StatusController implements StatusTemplateApi  {
+
+    private final StatusService statusService;
+
+    public StatusController(StatusService statusService) {
+        this.statusService = statusService;
+    }
+
+    @Override
+    public Mono<ResponseEntity<String>> getStatus(final ServerWebExchange exchange) {
```
Comment (AndreaCimini90) [inline src/main/java/it/pagopa/pn/bff/rest/StatusController.java:22]: Why the use of Mono?

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: docs/openapi/api-external-pn-bff.yaml (hunk 2 above)
Comment (alessandrogelmi) [reply docs/openapi/api-external-pn-bff.yaml:37]: I think to handle errors. It's the same on pn-delivery-push

File: docs/openapi/api-internal-pn-bff.yaml (hunk 3 above)
Comment (alessandrogelmi) [reply docs/openapi/api-internal-pn-bff.yaml:37]: I think to handle errors. It's the same on pn-delivery-push

File: src/main/java/it/pagopa/pn/bff/rest/StatusController.java (hunk 4 above)
Comment (alessandrogelmi) [reply src/main/java/it/pagopa/pn/bff/rest/StatusController.java:22]: A Mono object represents a single or empty value. Also on pn-delivery-push is used

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**